pkg/controller/certificates/signer/signer.go

```diff
@@ -87,6 +100,14 @@ func newSigner(caFile, caKeyFile string, client clientset.Interface, certificate
 		client:     client,
 		certTTL:    certificateDuration,
 	}
+
+	if chainFile != "" {
+		chainProvider, err := dynamiccertificates.NewDynamicCAContentFromFile("csr-controller", chainFile)
```

Comment:
> no intermediate expires after its parent.

I'm not sure we should impose this on users - apparently the x509 validation spec doesn't require validity window nesting. i'm pretty sure this restriction would be a problem for my cluster

Otherwise, I agree in principle with these validity checks, although I wonder how much validation is valuable here - this is an optional feature to append a set of certs to the signed cert. If you put certs here that don't form a chain, I'm not sure it would do much damage (eg, doesn't the Go tls server just ignore pointless intermediates?). We should definitely check that the certs can be parsed correctly as x509, which this code already does

I'll move to using `ClusterSigningCertFile` in which case we can take advantage of the existing cert loader

Reply:
As above, I think we may need to move back to a bool for whether we include signers, and then we can tackle multiple signing intermediates separately? in the simple case, we only need to provide one signer to the client, so none of this verification is needed

Reply:
I'm not in favor of the "include signer" bool that special-cases a single intermediate signer. I agree that it is important for the signer and intermediate chain to stay in sync.

Reply:
> that special-cases a single intermediate signer

does it? what if its `includesigners` and there can be a whole chain of signers?